sketch/sketch.ino

```diff
@@ -1,363 +1,486 @@
 
-    if (currentSelectionMode == Knob) {
-      // Read knob direction and change player selection
-      int knobDirection = readKnobDirection();
-      Serial.print("knobDirection: ");
-      Serial.print(knobDirection);
-      Serial.print(" - currentHighlightedPlayer: ");
-      if (knobDirection != 0) {
-        doPlayerSelection_updateHighlightedPlayer(knobDirection);
-      }
-      Serial.println(currentHighlightedPlayer);
-    } else {
-      if (digitalRead(redButtonPin) == LOW) {
-        doPlayerSelection_updateHighlightedPlayer(1); // Select clockwise active player
-        Serial.println("Player selection updated via button");
-      }
-    }
+// LEDs
+const int playerLedPins[] = {2,3,4,5}; // Need to add more pins if more players wanted.
+PlayerLed* playerLeds = new PlayerLed[PLAYER_COUNT];
+const int ledBlinkDuration = 300;
+const int ledIntervalDuration = 500;
 
-    if (digitalRead(greenButtonPin) == LOW) {
-      break;
-    }
-    // Read shoot button press, call logic and then exit loop if pressed
-    // Note: Probably more elegant to change state and handle animation/sounds in new state.
+// Buzzer
+const int buzzerPin = 6;
+struct Buzzer buzzer = {buzzerPin, 0};
 
-    delay(100); //  This delay value important and can affect selection knob accuracy.
-  }
+//  Button
+const int buttonAPin = 7;
+const int buttonBPin = 8;
+const int buttonInterval = 200; // number of millisecs between button readings
 
-  
-}
+struct Button buttonA = {buttonAPin, 0, buttonInterval, false};
+struct Button buttonB = {buttonBPin, 0, buttonInterval, false};
 
-void doPlayerTrigger() {
-      const auto shoot = playerChambersIndexes[currentHighlightedPlayer];
-      const auto result = playerChambers[currentHighlightedPlayer][shoot];
-      if (result == 1)
-      {
-          killPlayer(currentHighlightedPlayer);
-          return;
-      }
-      playerChambersIndexes[currentHighlightedPlayer] = shoot + 1;
-}
+// Potentiometer
+const int knobPin = A0;
+
+
+// Time
+unsigned long currentMillis = 0;    // stores the value of millis() in each iteration of loop()
+
+// END ====== GLOBAL STUFF ====================
 
-void killPlayer(int playerIndex) {
-  byte playerMask = ~(1 << playerIndex);
-  activePlayers &= playerMask;
-  Serial.println("ActivePlayers " + String(activePlayers));
+
+// Player state vars
+const int playerCount = 4; // Total players playing
+int activePlayerCount = 4; // Players that are alive
+byte activePlayers = 0b1111; // Byte with lower 4 bits representing which players are still alive
+int playerChambers[playerCount][6]; // Represent each player's chamber
+int playerChambersIndexes[playerCount] = {};
+
+// Potentiometer
+int previousKnobRead = 0;
+boolean knobReadSet = false;
+int knobSensitivity = 50;
+
+// Selection input type
+enum SelectionMode: int {
+  Knob,
+  Button
+};
+
+SelectionMode currentSelectionMode = Button; // Change this depending on secondary input type
+
+// GameState
+enum GameState: int {
```

Comment:
enum class GameState: int {
...
};
Will prevent putting `int` where you want GameState.